### PR TITLE
[DOCS] Fix `type` response values for index recovery API

### LINKED ISSUE #80264: [docs] Index recovery API Response wrong types
The Index recovery API documentation (https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-recovery.html#index-recovery-api-response-body) provides incorrect information. The valid values are listed as:

> STORE
The recovery is related to a node startup or failure. This type of recovery is called a local store recovery.
SNAPSHOT
The recovery is related to a snapshot restoration.
REPLICA
The recovery is related to a primary shard replication.
RELOCATING
The recovery is related to the relocation of a shard to a different node in the same cluster.

But this is not correct. These should be:
> EMPTY_STORE
EXISTING_STORE
PEER
SNAPSHOT
LOCAL_SHARDS.

### PR DESCRIPTION
**Pull request by jrodewig**
We updated the `type` response values in https://github.com/elastic/elasticsearch/pull/19516. This updates the docs with the correct values.

Closes https://github.com/elastic/elasticsearch/issues/80264

### Preview
https://elasticsearch_81000.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/indices-recovery.html#index-recovery-api-response-body

### CODE REVIEW

File: docs/reference/indices/recovery.asciidoc [hunk 1]
```diff
@@ -80,26 +80,27 @@ ID of the shard.
...
+`EMPTY_STORE`::
+An empty store. Indicates a new primary shard or the forced allocation of an
+empty primary shard using the <<cluster-reroute,cluster reroute API>>.
 
-`SNAPSHOT`::
-The recovery is related to
-a <<snapshots-restore-snapshot,snapshot restoration>>.
+`EXISTING_STORE`::
+The store of an existing primary shard. Indicates recovery is related
+to node startup, node failure, or the allocation of an existing primary shard.
```
Comment (DaveCTurner) [inline docs/reference/indices/recovery.asciidoc:91]: Typically not related to node failure, we'd usually promote a replica to primary in that case which doesn't involve recovery.

```suggestion
to node startup or the allocation of an existing primary shard.
```

File: docs/reference/indices/recovery.asciidoc [hunk 2]
```diff
@@ -80,26 +80,27 @@ ID of the shard.
...
 
-`REPLICA`::
-The recovery is related to a primary shard replication.
+`LOCAL_SHARDS`::
+Shards of another index on the same node. Indicates recovery is related to
+a <<indices-shrink-index,shrink index>> operation.
```
Comment (DaveCTurner) [inline docs/reference/indices/recovery.asciidoc:95]: I think this can be shrink, split or clone.

**Review by DaveCTurner — COMMENTED** [review]
Looks good, I left a couple of comments.

**Comment by jrodewig** [comment]
Thanks for the comments, @DaveCTurner. This is ready for another look at your convenience.

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**